src/handlers/transfers/FxFulfilService.js

```diff
@@ -307,17 +307,19 @@ class FxFulfilService {
 
   async processFxFulfil({ transfer, payload, action }) {
     await this.FxTransferModel.fxTransfer.saveFxFulfilResponse(transfer.commitRequestId, payload, action)
-    const cyrilOutput = await this.cyril.processFxFulfilMessage(transfer.commitRequestId, payload)
+    if (!transfer.fxWatchListId) {
+      throw new Error(`Commit request ID ${transfer.commitRequestId} not found in watch list`)
+    }
```

Comment:
Can we still have a function in cyril 'processFxFulfilMessage' for this check? I mean. you can remove the call 'getAllDetailsByCommitRequestId' in that function, but I would recommend the queries (getItemInWatchListByCommitRequestId) on watchlist in the cyril functions instead of having a join in other place. Because the watchlist logic may change and I wouldn't suggest merging watchList in the places other than cyril. I know it's additional query but I hope it will not impact performance that much compared to left join.